### ISSUE TITLE
Reload rules gracefully?

### ISSUE THREAD

**Issue by anitgandhi** [SpiderLabs/ModSecurity-nginx#94]
This is related to this issue https://github.com/SpiderLabs/ModSecurity/issues/1417 but I feel it's more appropriate in this repo.

If my nginx config has something like 

```
server {
    listen       80;

    # Turn on modsecurity
    modsecurity on;
    modsecurity_rules_file /etc/nginx/modsec/main.conf;
...
}
```

If I change any of the modsec language rules in `/etc/nginx/modsec/main.conf` or any of the subsequent modsec `Include` directives, is there a graceful way of reloading nginx to pick those up?

At the moment, sending a HUP to nginx master process (the normal way to tell nginx that there's a new config) won't do anything because from nginx's perspective, its config has not changed, so it doesn't pick up any of the new modsec rules. I've searched around but haven't found a way to force nginx to load in config as if it was for the first time. 

A full nginx stop and nginx start is ok but not ideal since it's not graceful and may cause downtime.

**Comment by victorhora** [comment]
@anitgandhi 

Have you tested with _nginx -s reload_ or [these](https://nginx.org/en/docs/control.html)?

Also you may want to try with the [usage examples](https://github.com/SpiderLabs/ModSecurity-nginx/blob/master/README.md) to try injecting ModSecurity configuration directives on different contexts such as _location_.

**Comment by anitgandhi** [comment]
Yes I did, `nginx -s reload` and sending a HUP are the same thing, that's what I tried already. 

I don't need location specific rules, the regular stuff works fine. 

I guess my question boils down to: is there a way to tell the nginx modsecurity connector to reload the rules explicitly, or possibly even a way for Nginx itself to tell its dynamic modules (modsecurity-nginx in this case) to reload its rules?
